src/main/java/org/jboss/mjolnir/client/component/administration/UserDialog.java

```diff
@@ -0,0 +1,220 @@
+                if (user == null || user.getName() == null) { // username is not modifiable for existing users
+                    userToSave.setName(kerberosNameBox.getText());
+                }
+                userToSave.setGithubName(gitHubNameBox.getText());
+                userToSave.setAdmin(adminCheckBox.getValue());
+                userToSave.setWhitelisted(whitelistedCheckBox.getValue());
+
+                // save
+                administrationService.editUser(userToSave, new AsyncCallback<EntityUpdateResult<KerberosUser>>() {
+                    @Override
+                    public void onFailure(Throwable caught) {
+                        ExceptionHandler.handle("Couldn't save user.", caught);
+                    }
+
+                    @Override
+                    public void onSuccess(EntityUpdateResult<KerberosUser> result) {
+                        if (result.isOK()) {
+                            if (user != null) {
+                                user.setName(userToSave.getName());
+                                user.setGithubName(userToSave.getGithubName());
+                            }
+                            onSave(userToSave);
+                            UserDialog.this.hide();
+                            UserDialog.this.removeFromParent();
+                        } else {
+                            feedbackLabel.setHTML(HTMLUtil.toUl(result.getValidationMessages()));
+                        }
+                    }
+                });
+            }
+        });
+
+        cancelButton.addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                UserDialog.this.hide();
+                UserDialog.this.removeFromParent();
+            }
+        });
+    }
+
+    private void initRegisterDialog() {
+        setText("Register User");
+
+        kerberosNameBox.addChangeHandler(new ChangeHandler() {
+            @Override
+            public void onChange(ChangeEvent event) {
+                setActiveKrbAccount(kerberosNameBox.getText());
+            }
+        });
+
+        submitButton.addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                // create new user instance
+                final KerberosUser user = new KerberosUser();
+
+                user.setName(kerberosNameBox.getText());
+                user.setGithubName(gitHubNameBox.getText());
+                user.setAdmin(adminCheckBox.getValue());
+                user.setWhitelisted(whitelistedCheckBox.getValue());
+
+                // save
+                administrationService.registerUser(user, new AsyncCallback<EntityUpdateResult<KerberosUser>>() {
+                    @Override
+                    public void onFailure(Throwable caught) {
+                        ExceptionHandler.handle("Couldn't save user. Check whether the kerberos username is unique.", caught);
```

Comment:
This should be more specific - if there is expected error condition, like username already exists, it should be distinguishable from other error conditions. Now expected and unexpected error conditions are handled in a same way and there is no way to tell what actually happened, so we have to guess.

In this case it should be specifically checked on backend if the username already exists before saving, and if so specific validation message could be returned via EntityUpdateResult which would be handled by onSuccess() method bellow.